### PR TITLE
Kill YARN jobs with a loop to catch any late starting jobs

### PR DESCRIPTION
**Pull request by jschmidt10**
In addition to the retry loop, I removed the previous attempt which introduced the pid-functions.sh. That's no longer needed.

### CODE REVIEW

File: warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh [hunk 1]
```diff
@@ -19,6 +19,56 @@ fi
...
+INGEST_JOB_POSTKILL_DELAY_SECS="${INGEST_JOB_POSTKILL_DELAY_SECS}"
+INGEST_JOB_KILL_MAX_RETRIES="${INGEST_JOB_KILL_MAX_RETRIES}"
+INGEST_JOB_KILL_LOOP_DELAY_SECS="${INGEST_JOB_KILL_LOOP_DELAY_SECS}"
+
+[[ -n "${INGEST_JOB_POSTKILL_DELAY_SECS}" ]] || { echo "Must define INGEST_JOB_POSTKILL_DELAY_SECS"; exit 1; }
+[[ -n "${INGEST_JOB_KILL_MAX_RETRIES}" ]] || { echo "Must define INGEST_JOB_KILL_MAX_RETRIES"; exit 1; }
+[[ -n "${INGEST_JOB_KILL_LOOP_DELAY_SECS}" ]] || { echo "Must define INGEST_JOB_KILL_LOOP_DELAY_SECS"; exit 1; }
+
+function getNumRunningJobs() {
+  echo "$(jps -lm | grep -c datawave.ingest.mapreduce.job.IngestJob)"
```
Comment (matthpeterson) [inline warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh:31]:
```
echo "$(cmd)"
```
can be replaced with 
```
cmd
```

File: warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh [hunk 2]
```diff
@@ -19,6 +19,56 @@ fi
...
+  sleep "${INGEST_JOB_POSTKILL_DELAY_SECS}"
+}
+
+function startYarnKillLoop() {
+  killYarnJobsAndWait
+
+  numRunningJobs=$(getNumRunningJobs)
+  retryNum=1
+
+  while [[ ${numRunningJobs} > 0 && ${retryNum} < ${INGEST_JOB_KILL_MAX_RETRIES} ]]; do
```
Comment (matthpeterson) [inline warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh:49]: I'm not sure why ```>``` is for string comparisons in bash, but ```-gt``` is the correct operator for this.  Same observation for ```<``` and ```-lt```.

```
$ retryNum=00
$ if [[ ${numRunningJobs} > 0 ]] ; then echo "00 not really greater than 0"; fi
00 not really greater than 0
$ if [[ ${numRunningJobs} -gt 0 ]] ; then echo "00 not really greater than 0"; fi
```

Comment (hlgp) [reply warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh:49]: +1

File: warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh [hunk 3]
```diff
@@ -19,6 +19,56 @@ fi
...
+      echo "${numRunningJobs} jobs still running. Waiting for ${INGEST_JOB_KILL_LOOP_DELAY_SECS} seconds before issuing the next yarn kill."
+      sleep "${INGEST_JOB_KILL_LOOP_DELAY_SECS}"
+    fi
+
+    numRunningJobs=$(getNumRunningJobs)
+    retryNum=$(( retryNum + 1 ))
+  done
+
+  numRunningJobs=$(getNumRunningJobs)
+  if [[ ${numRunningJobs} > 0 ]]; then
```
Comment (hlgp) [inline warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh:65]:
```suggestion
  if [[ ${numRunningJobs} -gt 0 ]]; then
```

File: warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh (hunk 2 above)
Comment (hlgp) [inline warehouse/ingest-scripts/src/main/resources/bin/ingest/stop-ingesters.sh:49]:
```suggestion
  while [[ ${numRunningJobs} -gt 0 && ${retryNum} -lt ${INGEST_JOB_KILL_MAX_RETRIES} ]]; do
```

**Comment by jschmidt10** [comment]
This is stale and possibly OBE.

**Pull request closed without merging**